### PR TITLE
Fr 1855 pdf conversion

### PR DESCRIPTION
**Pull request by ocarleton**
**Before creating a pull request make sure that:**

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/controllers/HearingOrderController.java [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+
+    }
+
+    private CaseDocument getLatestHearingOrderAsPdf(Map<String, Object> caseData, String authorisationToken) {
+        CaseDocument hearingOrderDocument = documentHelper.getLatestContestedDraftOrderCollection(caseData);
+        if (hearingOrderDocument == null) {
+            throw new InvalidCaseDataException(BAD_REQUEST.value(), "Missing data from callbackRequest.");
+        }
+
+        log.info("ABX {}", hearingOrderDocument.getDocumentFilename().toLowerCase());
```
Comment (shaunganley) [inline src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/controllers/HearingOrderController.java:84]: this testing logs?

**Review by shaunganley — APPROVED** [review]

File: src/test/java/uk/gov/hmcts/reform/finrem/caseorchestration/controllers/HearingOrderControllerTest.java [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+    }
+
+    @Test
+    public void convertPdfDocument() throws Exception {
+        when(genericDocumentService.stampDocument(any(), anyString())).thenReturn(getCaseDocument());
+        when(moveCollectionService.moveCollection(anyMap(), anyString(), anyString()))
+            .thenReturn(new HashMap<String, Object>() {});
+        when(documentHelper.getLatestContestedDraftOrderCollection(anyMap())).thenReturn(getCaseDocument());
+
+        mvc.perform(post(CONVERSION_URL)
```
Comment (shaunganley) [inline src/test/java/uk/gov/hmcts/reform/finrem/caseorchestration/controllers/HearingOrderControllerTest.java:58]: call methods directly in unit tests. add a functional test to call the endpoint via HTTP

File: charts/finrem-cos/values.yaml [hunk 3]
```diff
@@ -7,7 +7,7 @@ java:
     AUTH_PROVIDER_SERVICE_CLIENT_BASEURL : "http://rpe-service-auth-provider-{{ .Values.global.environment }}.service.core-compute-{{ .Values.global.environment }}.internal"
     IDAM_API_URL: "https://idam-api.{{ .Values.global.environment }}.platform.hmcts.net"
     IDAM_S2S_URL : "http://rpe-service-auth-provider-{{ .Values.global.environment }}.service.core-compute-{{ .Values.global.environment }}.internal"
-    DOCUMENT_GENERATOR_SERVICE_API_BASEURL : "http://finrem-dgcs-{{ .Values.global.environment }}.service.core-compute-{{ .Values.global.environment }}.internal"
+    DOCUMENT_GENERATOR_SERVICE_API_BASEURL : "http://finrem-dgcs-pr-91.service.core-compute-preview.internal"
```
Comment (jakub-kozlowski) [inline charts/finrem-cos/values.yaml:10]: This is for testing probably - remember to fix?

File: src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/controllers/HearingOrderController.java [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+        @ApiResponse(code = 200, message = "Callback was processed successfully or in case of an error message is attached to the case",
+            response = AboutToStartOrSubmitCallbackResponse.class),
+        @ApiResponse(code = 400, message = "Bad Request"),
+        @ApiResponse(code = 500, message = "Internal Server Error")})
+    public ResponseEntity<AboutToStartOrSubmitCallbackResponse> storeHearingOrder(
+        @RequestHeader(value = AUTHORIZATION_HEADER) String authorisationToken,
+        @NotNull @RequestBody @ApiParam("CaseData") CallbackRequest callback) {
+
+        validateCaseData(callback);
+        log.info("storing hearing order");
```
Comment (jakub-kozlowski) [inline src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/controllers/HearingOrderController.java:66]: add some details to the log, at least case ID. Will be not much use w/o

File: src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/service/MoveCollectionService.java [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+
+import org.springframework.stereotype.Service;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+
+@Service
+public class MoveCollectionService {
```
Comment (jakub-kozlowski) [inline src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/service/MoveCollectionService.java:11]: I'm not a fan of using a service when it's not required. What benefit comes with making a helper method a service? A price to pay is bigger spring context to load. If it comes w/o benefit it seems questionable.

**Review by jakub-kozlowski — APPROVED** [review]

**Pull request closed without merging**